Tests failed on typechecking:


> nps is executing `test.typecheck` : tsc -p tsconfig.json
__tests__/test-fetch.js(53,9): error TS2322: Type '{ request: (request: any) => Promise<{ statusCode: number; statusMessage: string; headers: {}; body: never[]; }>; }' is not assignable to type 'HttpClient'.
  The types returned by 'request(...)' are incompatible between these types.
    Type 'Promise<{ statusCode: number; statusMessage: string; headers: {}; body: never[]; }>' is not assignable to type 'Promise<GitHttpResponse>'.
      Property 'url' is missing in type '{ statusCode: number; statusMessage: string; headers: {}; body: never[]; }' but required in type 'GitHttpResponse'.
The script called "test.typecheck" which runs "tsc -p tsconfig.json" failed with exit code 2 https://github.com/sezna/nps/blob/master/other/ERRORS_AND_WARNINGS.md#failed-with-exit-code
The script called "test" which runs "nps lint && nps build.test && nps test.typecheck && nps test.setup && nps test.node && nps test.chrome && nps test.teardown" failed with exit code 2 https://github.com/sezna/nps/blob/master/other/ERRORS_AND_WARNINGS.md#failed-with-exit-code

The error came from `HttpClient` but the issue came from this PR.

--- a/__tests__/test-fetch.js
+++ b/__tests__/test-fetch.js
@@ -27,6 +27,7 @@ describe('fetch', () => {
     })
     const requests = []
     const onAuthArgs = []
+    const emptyBody = async function* () {}
     const mockHttp = {
       request: async request => {
         requests.push(request)
@@ -36,14 +37,14 @@ describe('fetch', () => {
               statusCode: 401,
               statusMessage: 'Unauthorized',
               headers: {},
-              body: [],
+              body: emptyBody(),
             }
           : {
               url: request.url,
               statusCode: 403,
               statusMessage: 'Forbidden',
               headers: {},
-              body: [],
+              body: emptyBody(),
             }
       },
     }

--- a/__tests__/test-fetch.js
+++ b/__tests__/test-fetch.js
@@ -32,12 +32,14 @@ describe('fetch', () => {
         requests.push(request)
         return requests.length === 1
           ? {
+              url: request.url,
               statusCode: 401,
               statusMessage: 'Unauthorized',
               headers: {},
               body: [],
             }
           : {
+              url: request.url,
               statusCode: 403,
               statusMessage: 'Forbidden',
               headers: {},

